src/Symfony/Component/Messenger/MessageBus.php

```diff
@@ -55,6 +55,9 @@ private function callableForNextMiddleware(int $index): callable
         $middleware = $this->indexedMiddlewares[$index];
 
         return function ($message) use ($middleware, $index) {
+            // Do not provide the envelope if the middleware cannot read it:
+            $message = Envelope::wrap($message)->openFor($middleware);
```

Comment:
Just for the readbility of not overriding `$message`, what about:
```php
return $middleware->handle(
    Envelope::wrap($message)->openFor($middleware),
    $this->callableForNextMiddleware($index + 1)
);
```

Reply:
Actually, now that only middlewares are implementing this interface in the core, I removed the `Envelope::openFor()` method and inlined the checks in the bus directly.